pkg/cmd/ssh/ssh_patch_options.go

```diff
@@ -0,0 +1,400 @@
+	if err != nil {
+		return nil, err
+	}
+
+	if bastion.ObjectMeta.UID == "" {
+		return nil, fmt.Errorf("Bastion '%s' in namespace '%s' not found", o.BastionName, shoot.Namespace)
+	}
+
+	return bastion, nil
+}
+
+func getAuthInfo(ctx context.Context, manager target.Manager) (*api.AuthInfo, error) {
+	currentTarget, err := manager.CurrentTarget()
+	if err != nil {
+		return nil, err
+	}
+
+	gardenTarget := target.NewTarget(currentTarget.GardenName(), "", "", "")
+
+	clientConfig, err := manager.ClientConfig(ctx, gardenTarget)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve client config for target garden %s: %w", currentTarget.GardenName(), err)
+	}
+
+	rawConfig, err := clientConfig.RawConfig()
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve raw config: %w", err)
+	}
+
+	context, ok := rawConfig.Contexts[rawConfig.CurrentContext]
+	if !ok {
+		return nil, fmt.Errorf("no context found for current context %s", rawConfig.CurrentContext)
+	}
+
+	authInfo, ok := rawConfig.AuthInfos[context.AuthInfo]
+	if !ok {
+		return nil, fmt.Errorf("no auth info found with name %s", context.AuthInfo)
+	}
+
+	return authInfo, nil
+}
+
+func getBastionsOfUser(ctx context.Context, gardenClient gardenClient.Client, authInfo *api.AuthInfo) ([]*gardenoperationsv1alpha1.Bastion, error) {
+	var bastionOfUser []*gardenoperationsv1alpha1.Bastion
+
+	currentUser, err := getCurrentUser(ctx, gardenClient, authInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	list, err := gardenClient.ListBastions(ctx)
+	if err != nil || len(list.Items) == 0 {
+		return nil, err
+	}
+
+	for i := range list.Items {
+		bastion := list.Items[i]
+		if createdBy, ok := bastion.Annotations["gardener.cloud/created-by"]; ok {
+			if createdBy == currentUser {
+				bastionOfUser = append(bastionOfUser, &bastion)
+			}
+		}
+	}
+
+	return bastionOfUser, nil
+}
+
+func patchBastionIngress(ctx context.Context, o *SSHPatchOptions, gardenClient gardenClient.Client, currentTarget target.Target) error {
+	var policies []gardenoperationsv1alpha1.BastionIngressPolicy
+
+	oldBastion := o.Bastion.DeepCopy()
+
+	for _, cidr := range o.CIDRs {
+		if *o.Bastion.Spec.ProviderType == "gcp" {
+			ip, _, err := net.ParseCIDR(cidr)
+			if err != nil {
+				return err
+			}
+
+			if ip.To4() == nil {
+				if !o.AutoDetected {
+					return fmt.Errorf("GCP only supports IPv4: %s", cidr)
+				}
+
+				fmt.Fprintf(o.IOStreams.Out, "GCP only supports IPv4, skipped CIDR: %s\n", cidr)
+
+				continue // skip
+			}
+		}
+
+		policies = append(policies, gardenoperationsv1alpha1.BastionIngressPolicy{
+			IPBlock: networkingv1.IPBlock{
+				CIDR: cidr,
+			},
+		})
+	}
+
+	if len(policies) == 0 {
+		return errors.New("no ingress policies could be created")
+	}
+
+	o.Bastion.Spec.Ingress = policies
+
+	return gardenClient.PatchBastion(ctx, o.Bastion, oldBastion)
+}
+
+func getManagerTargetAndGardenClient(f util.Factory) (target.Manager, target.Target, gardenClient.Client, error) {
```

Comment:
This is a strange reuse function. I would inline the code

Reply:
+1
I hoped this one would vanish after putting the needed properties on the options in the complete function (inlined)